File: articles/grace-hopper-compilers-and-the-humanization-of-programming/prolog/compiler_pipeline.pl
token(add).
token(payroll_total).
token(to).
token(tax_base).

pipeline(source, "ADD PAYROLL-TOTAL TO TAX-BASE").
pipeline(tokens, [add, payroll_total, to, tax_base]).
pipeline(target_code, "machine-specific instruction sequence").

% Query example:
% ?- pipeline(tokens, Tokens).

File: articles/grace-hopper-compilers-and-the-humanization-of-programming/rust/compiler_pipeline.rs
fn main() {
    let source = "ADD PAYROLL-TOTAL TO TAX-BASE";
    let tokens: Vec<&str> = source.split_whitespace().collect();
    println!("source={}", source);
    println!("tokens={:?}", tokens);
    println!("target_code=machine-specific instruction sequence");
}
